### PR TITLE
Add some data filtering support

### PR DESCRIPTION
**Pull request by cbaines**
This branch includes some changes I made when extracting out a dump of part of the content tagged to the Topic Taxonomy.

### CODE REVIEW

**Comment by koetsier** [comment]
could you add a description to this PR?

File: python/data/taxons.py [hunk 1]
```diff
@@ -27,3 +27,11 @@ def content_item_tagged_to_topic_taxonomy(content_item):
             content_item_taxons(content_item)
         )
     )
+
+def content_item_within_part_of_taxonomy(content_item, taxonomy_part_content_id):
```
Comment (koetsier) [inline python/data/taxons.py]: It would be nice to have some tests in this module

File: python/data/__init__.py [hunk 2]
```diff
@@ -23,3 +25,13 @@ def items_from_content_file(datadir=None, filename="content.json.gz"):
 
     with gzip.open(full_filename, mode='rt') as content_file:
         yield from ijson.items(content_file, prefix='item')
+
+def write_content_file(filename, content, datadir=None):
+    if datadir is None:
+        datadir = os.getenv("DATADIR")
```
Comment (1pretz1) [inline python/data/__init__.py]: Probably would have to add `datadir = os.getenv("DATADIR") or "data"` here to make sure there is a default if the `DATADIR` variable is not set.

File: python/data/taxons.py (hunk 1 above)
Comment (cbaines) [reply python/data/taxons.py]: I've finally got around to having a look at this (see defe95a ), let me know what you think :)

**Comment by cbaines** [comment]
> could you add a description to this PR?

Sure, I've added a short one, but let me know if this doesn't answer any questions that you have.

File: python/data/__init__.py (hunk 2 above)
Comment (cbaines) [reply python/data/__init__.py]: Good spot, I've added a default now :+1:

**Pull request merged**